src/orchestrator.ts

```diff
@@ -0,0 +1,64 @@
+import { ethers } from "ethers";
+import { Configuration } from "./types";
+import { FileManager } from "./file-manager";
+import { BlockFinder } from "./block-finder";
+import { DistributorDetector } from "./distributor-detector";
+import { BalanceFetcher } from "./balance-fetcher";
+import { RecipientRecievedScanner } from "./recipient-recieved-scanner";
+import { FeeCalculator } from "./fee-calculator";
+
+export async function orchestrate(config: Configuration): Promise<void> {
+  // Initialize infrastructure
+  const fileManager = new FileManager(config.storeDirectory);
+  const provider = new ethers.JsonRpcProvider(config.rpcUrl);
+  fileManager.ensureStoreDirectory();
+
+  // Parse date range
+  const { startDate, endDate } = parseDateRange(config);
+
+  // Execute pipeline components sequentially
+  await executePipeline(fileManager, provider, startDate, endDate);
+}
+
+function parseDateRange(config: Configuration): {
+  startDate: Date;
+  endDate: Date;
+} {
+  const today = new Date();
+  today.setUTCHours(0, 0, 0, 0);
+
+  const startDate = config.startDate ? new Date(config.startDate) : today;
```

Comment:
Shouldn't we start from the date when the chain started, if no one specifies a date?